src/dialog/dialog.service.ts

```diff
@@ -193,17 +193,28 @@ export class DialogService {
 	}
 
 	/**
-	 * Closes all known `Dialog`s. Does not focus any previous elements, since we can't know which would be correct
-	 */
-	closeAll() {
-		DialogService.dialogRefs.forEach(ref => ref.destroy());
-		DialogService.dialogRefs.clear();
-		DialogService.dialogCloseSubscription.unsubscribe();
-		this.isClosed.emit(true);
+	 * Closes all known `Dialog`s. Does not focus any previous elements, since we can't know which would be correct.
+	 * If `preventClosure` parameter is provided, it closes all know `Dialog`s except the given Dialog ref.
+	 * @param preventClosure the dialogRef to keep open while all others are closed
+	 */
+	closeAll(preventClosure?: ComponentRef<Dialog>) {
+		// DialogService.dialogRefs.forEach(ref => ref.destroy());
```

Comment:
On one hand we could probably remove this, on the other it's a pretty handy description of whats going on below, so 🤷‍♂

Reply:
I meant to remove it before I committed and pushed 🤦‍♂ but you're right it can be a pretty good description!

Reply:
On a side note... if there are two tooltips with a click trigger and there is no element within the dialog to focus it automatically focuses on the other element when you change which is opened. Not sure if that is intended but if you click one and press enter on the other while the other tooltip is focused it keeps both of them open and unable to close.

Reply:
Can you demo this to me? It seems like a bug ... but it could also just be an awkward interaction pattern